Feita alterações na rota users/update e criada a rota game/update

1995-Matheus-Lima opened the pull request:
- Feita as alterações sugeridas em user/Update
- Prototipada a  função update de game/Update -Aguardarei a criação das funções create para fazer mais testes nessa função-
- a função async em db.js para checar a conexão do modo em que estava declarada estava causando diversos error no meu pc, por isso propus uma alteração de modo em que o erro parou e fica bem claro o que se deseja fazer "checkConnection"

1995-Matheus-Lima committed:
Feita alterações na rota users/update e criada a rota game/update

pedro-de-oliveira-guedes reviewed (changes requested):
# Code Review
Opa, boa tarde Matheus! Tudo bem contigo?
Muito boas as alterações feitas cara, acertou na mosca! 🎯🥇😉
Correndo o risco de ser chato, vou te pedir só mais uma mínima alteração no seu código, pode ser? 😬
Vou tratar os arquivos de rotas como um só, já que a implementação é quase a mesma. 🧐

---
## Arquivo db.js
Muito boa a sacada de dar nome para a função de autenticação do banco de dados! Como é um arquivo meio "enterrado" e que não é acessado frequentemente, não me preocupei muito com dar um nome à função.
Pode deixar assim, que está ótimo! 😉
Parabéns Matheus! 🏆🏅

---
## Arquivos ```gameRoutes.js``` & ```userRoutes.js```
Tá no caminho certinho Matheus, só tem um detalhe...
Nós precisamos utilizar o ```error middleware``` definido ao final do programa para que ele seja válido. 😔

Mas calma, não é nada muito difícil 🧐. Eu inclusive adicionei dois pacotes para tornar o tratamento de erros mais fácil. 😉
Observe que todos os erros que você fez o tratamento são erros do cliente (código 4XX). Seja porque ele tentou acessar uma tupla ou propriedade que não existe, ou simplesmente porque errou o corpo da requisição. Mas e quando o erro não é do cliente 🤔?  Quando simplesmente há um erro, mas ele não foi causado por falha humana 🤔? Precisamos também tratar os erros por parte do servidor! 🧐

<blockquote>

Para saber mais sobre erros HTTP e seus respectivos códigos, você pode passar o olho por este artigo na internet:
https://developer.mozilla.org/pt-BR/docs/Web/HTTP/Status

</blockquote>

Para isso, utilizaremos uma função importada na linha 5 de ambos arquivos, a ```asyncHandler```. Ela é necessária porque os erros do servidor muitas vezes (praticamente sempre) não são tão rápidos quanto a execução do programa, por isso são assíncronas. Esta função permite que erros nesse tipo de programa sejam tratados independentemente da execução não ser síncrona.
```js
router.get('/read_all/', asyncHandler (async (req, res, next) => {
    try {
        const users = await Usuarios.findAll();
        res.status(200).json(users);
    }
    catch (error) {
        next(error);
        return;
    }
}));
```
O trecho de código acima foi utilizado por mim para implementar minha rota. Por favor, utilize-o como exemplo para desenvolver sua rota.

Outro detalhe que pode ser utilizado para melhorar o manejamento de erros, é a função importada na linha 6 de ambos os arquivos, a ```createError()```. Nela é possível informar um código (5XX para servidor) e o erro ocorrido, entre outras coisas. O código anterior ficaria então:
```js
router.get('/read_all/', asyncHandler (async (req, res, next) => {
    try {
        const users = await Usuarios.findAll();
        res.status(200).json(users);
    }
    catch (error) {
        next(createError(500, error));
        return;
    }
}));
```
---
## Conclusão
Além do pequeno detalhe a ser adicionado, não há nenhum erro aparente no programa. O código está muito bom Matheus! 😉
Parabéns, continue assim!!! 🚀🥳

1995-Matheus-Lima committed:
adicionado tratamento de erros

pedro-de-oliveira-guedes committed:
Merge branch 'main' into Update_Dev

pedro-de-oliveira-guedes committed:
Retirando abreviações de palavras em inglês.

Quando enviamos uma resposta para o cliente, caracteres especiais tais como " ' ", "\n" entre outros, não são renderizados da forma que se espera (pelo menos não em todos os serviços de API). Isso ocasiona respostas difíceis de ler e com ainda mais caracteres especiais na interpretação da resposta.

pedro-de-oliveira-guedes committed:
Retirando abreviações de palavras em inglês.

Quando enviamos uma resposta para o cliente, caracteres especiais tais como " ' ", "\n" entre outros, não são renderizados da forma que se espera (pelo menos não em todos os serviços de API). Isso ocasiona respostas difíceis de ler e com ainda mais caracteres especiais na interpretação da resposta.

pedro-de-oliveira-guedes reviewed (approved):
# Code Review
Opa, boa noite Matheus! Tudo tranquilo com você? ☕
Obrigado pela paciência em aguardar a Code Review, vou me esforçar para fazer mais rápido das próximas vezes! 😉
O código tá sensacional, parabéns! 🥳🎉
Já vou incluir aqui! Continue assim! 😉🥇🏆
